File: codeforces/ungrouped/cf2024b.py
def solve(n, k, a):
    a = list(a)
    a.sort(reverse=True)
    count = 0
    height = 0
    while k > 0:
        if k > (a[-1] - height) * len(a):
            k -= (a[-1] - height) * len(a)
            count += (a[-1] - height) * len(a)
            height = a[-1]
            while a[-1] == height:
                count += 1
                a.pop(-1)

        else:
            count += k
            k = 0
    return count


num_of_tc = int(input())
for _ in range(num_of_tc):
    n, k = map(int, input().split())
    a = map(int, input().split())
    print(solve(n, k, a))

File: codeforces/ungrouped/cf2024b.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int solve(int n, int k, vector<int>& a) {
    sort(a.begin(), a.end());
    int count = 0;
    int height = 0;

    while (k > 0) {
        if (k > (a[0] - height) * a.size()) {
            k -= (a[0] - height) * a.size();
            count += (a[0] - height) * a.size();
            height = a[0];

            while (!a.empty() && a[0] == height) {
                count++;
                a.erase(a.begin());
            }
        } else {
            count += k;
            k = 0;
        }
    }
    
    return count;
}

int main() {
    int num_of_tc;
    cin >> num_of_tc;

    for (int i = 0; i < num_of_tc; ++i) {
        int n, k;
        cin >> n >> k;
        vector<int> a(n);

        for (int j = 0; j < n; ++j) {
            cin >> a[j];
        }

        cout << solve(n, k, a) << endl;
    }

    return 0;
}
